cpu/sam3/periph/gpio.c

```diff
@@ -53,7 +53,7 @@
 /**
  * @brief Allocation of memory for 7 independent interrupt slots
  */
-static gpio_isr_ctx_t exti_ctx[CTX_NUMOF] = {{0}};
+static gpio_isr_ctx_t exti_ctx[CTX_NUMOF];
```

Comment:
I'm always uncertain: is it 120% safe to assume this will be initialised with all `\0`? Because the driver needs that to identify empty slots. I think with `static` is should be init with zero, but ...

Reply:
The C standard specifies that variables with static storage duration (static and global variables) are initialized to zero. Variables with automatic storage duration (local variables, stack allocated) are not.
http://www.open-std.org/jtc1/sc22/wg14/www/standards WG14/N1256 section 6.7.8.10
>If an object that has automatic storage duration is not initialized explicitly, its value is indeterminate. If an object that has static storage duration is not initialized explicitly, then:
>
> - if it has pointer type, it is initialized to a null pointer;
> - if it has arithmetic type, it is initialized to (positive or unsigned) zero;
> - if it is an aggregate, every member is initialized (recursively) according to these rules;
> - if it is a union, the first named member is initialized (recursively) according to these rules.

This array has static storage duration. In practice, the compiler will place this array in the .bss section in the executable, which is zeroed out during c library initialization (by the `__libc_init_array` call in newlib)